packages/client-config/src/client-config-schema/schema_v1.json

```diff
@@ -0,0 +1,401 @@
+{
```

Comment:
I thought I had left this comment in my original review, but looks like it must not have saved. Anyway, have we considered instead of generate the TS type from a json schema, to do the conversion the other way? We could use a package like https://www.npmjs.com/package/ts-json-schema-generator to do the conversion. My reasoning is it is much easier to author TS types than a JSON schema.

Reply:
TS to JSONSchema is a lossy conversion. Moreover since we want to generate types from the same source, it's much easier for all libraries to generate types from JSONSchema rather than TS (or TS -> JSONSchema -> others)